tests/features/layers/activation-order/test/testcase.cpp

```diff
@@ -0,0 +1,206 @@
+/* -*- mode: c++ -*-
+ * Copyright (C) 2020  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testing/setup-googletest.h"
+
+SETUP_GOOGLETEST();
+
+namespace kaleidoscope {
+namespace testing {
+namespace {
+
+using ::testing::IsEmpty;
+
+class LayerActivationOrder : public VirtualDeviceTest {
+ public:
+  const KeyAddr KEYSWITCH_TOP_LEFT = KeyAddr{0, 0};   // layer-dependent key
+  const KeyAddr KEYSWITCH_TOP_RIGHT = KeyAddr{0, 15}; // layer-dependent key
+  const KeyAddr KEYSWITCH_LEFT_PALM = KeyAddr{3, 6};  // ShiftToLayer(1)
+  const KeyAddr KEYSWITCH_RIGHT_PALM = KeyAddr{3, 9}; // ShiftToLayer(2)
+  const KeyAddr KEYSWITCH_LEFT_THUMB_RIGHTMOST = KeyAddr{3, 7}; // L0 deactivate macro
+
+  const Key LAYER0_KEY = Key_0;
+  const Key LAYER1_KEY = Key_1;
+  const Key LAYER2_KEY = Key_2;
```

Comment:
[begin nitpick]

The [coding style guide](https://github.com/keyboardio/Kaleidoscope/blob/master/docs/codebase/code-style.md#constant-names) would have us naming these constants things like `kKeyswitchTopLeft` instead, and reserving all-caps for preprocessor macros. I'm ambivalent about this, and actually tend to just name contants like other variables, but I would like to get a ruling on the preferred style from those in authority, and perhaps an update to the style guide.

On a separate note, I think it would be preferable for the first group to be in namespace `kaleidoscope::testing`, and to be defined like this:
```
constexpr KeyAddr KEYSWITCH_TOP_LEFT{0, 0};
```
…which would be a more C++ style brace initialization.

[end nitpick]

Reply:
The coding style guide is... still somewhat a work in progress - I need to open specific issues so we can track what needs to be updated still. We don't prefix constants with `k` anywhere in existing code, for example (outside of googletest).

> constexpr KeyAddr KEYSWITCH_TOP_LEFT{0, 0};

We'd need to make it static then, and then have a `constexpr KeyAddr LayerActivationOrder::KEYSWITCH_TOP_LEFT;` outside of the class too, and the same for every other constant. While it would be more C++, and more in line with how we usually do things in Kaleidoscope, it doesn't really make the code any more readable, imo.

Reply:
I'm well aware that the style guide isn't followed everywhere (or, in some cases, anywhere). I'm arguing that it would be real nice for someone who wants to follow a consistent style to have a decision from the people who are in authority, so I can stop having a mental debate every time I have to name a constant. And using all caps for constants in a file with lots of preprocessor macros is distracting.

Re: namespace: I'm saying that these constants aren't just for class `LayerActivationOrder`, so they should be defined outside the class. Preferably in a place (in the case of that first block) where they could be included by other tests, as well. It only takes one line to define each constant, in the `testing` namespace instead of in the class; nothing else would change.